src/model/board.py

```diff
@@ -0,0 +1,6 @@
+class Board:
```

Comment:
Here we can have a board composed of a matrix of tiles where each tile has type `typing.Union[Fruit, Snake, None]`

Maybe it is worth it to declare an alias like `Tile = Union[Fruit, Snake, None]`

Reply:
I haven't used aliases very much before. I will look into them also.